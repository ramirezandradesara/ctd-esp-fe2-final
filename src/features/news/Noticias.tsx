import { INoticiasNormalizadas } from './types/news.types'
import {
  TarjetaNoticia,
  FechaTarjetaNoticia,
  DescripcionTarjetaNoticia,
  ImagenTarjetaNoticia,
  TituloTarjetaNoticia,
  ContenedorNoticias,
  ListaNoticias,
  TituloNoticias,
  BotonLectura,
} from "./styled";
import { useNews } from "./hooks/useNews";
import useModal from "./hooks/useModal";
import Modal from "./modal/Modal";

const Noticias = () => {
  const noticias = useNews()
  const { modal, isOpenModal } = useModal();

  return (
    <ContenedorNoticias>
      <TituloNoticias>Noticias de los Simpsons</TituloNoticias>
      <ListaNoticias>
        {noticias?.map((noticia: INoticiasNormalizadas) => (
          <TarjetaNoticia>
            <ImagenTarjetaNoticia src={noticia.imagen} />
            <TituloTarjetaNoticia>{noticia.titulo}</TituloTarjetaNoticia>
            <FechaTarjetaNoticia>{noticia.fecha}</FechaTarjetaNoticia>
            <DescripcionTarjetaNoticia>
              {noticia.descripcionCorta}
            </DescripcionTarjetaNoticia>
            <BotonLectura onClick={() => isOpenModal(noticia)}>Ver m√°s</BotonLectura>
          </TarjetaNoticia>
        ))}

        {modal && <Modal isOpenModal={isOpenModal} modal={modal} />}
      </ListaNoticias>
    </ContenedorNoticias>
  );
};

export default Noticias;
