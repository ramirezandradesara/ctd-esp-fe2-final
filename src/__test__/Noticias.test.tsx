import { act, fireEvent, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import Noticias from '../features/news/Noticias'
import { customRender } from '../test-utils'

describe("Pruebas en <Noticias />", () => {

    let renderComponent = () => customRender(<Noticias />)

    test("Renderizado inicial con tÃ­tulo", async () => {
        renderComponent();

        expect(screen.getByText('Noticias de los Simpsons')).toBeInTheDocument();
    });

    test("Snapshot de <Noticias />", async () => {
        let { asFragment } = customRender(<Noticias />)
        let fragment = asFragment()

        expect(fragment).toMatchSnapshot();
    });
});
