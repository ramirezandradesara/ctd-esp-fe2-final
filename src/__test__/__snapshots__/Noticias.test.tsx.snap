// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Cita /> Snapshot de <Noticias /> 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe dTSoUE"
  >
    <h2
      class="sc-gswNZR jJQjGi"
    >
      Noticias de los Simpsons
    </h2>
    <div
      class="sc-dkrFOg eDTxVm"
    />
  </div>
</DocumentFragment>
`;
